sched/Kconfig

```diff
@@ -1937,3 +1937,14 @@ config GROUP_KILL_CHILDREN_TIMEOUT_MS
 		> 0 means wait timeout
 		= 0 means don't do kill signal
 
+config INITIAL_PID_LIMIT
```

Comment:
Maybe it makes sense to make this option a power of `2` instead of adding compile time check? The description will specify that actual limit will be `2^CONFIG_INITIAL_PID_LIMIT`. And init value can be as simple as `(1u <<  CONFIG_INITIAL_PID_LIMIT)`

Reply:
So shall rename of config as `xxx_EXPONENT` to match its semantics? that sounds complex.

Reply:
Log2 is not exponent based (is 2 based), so I do not think we need to add any exponent suffix. Let me think about proper naming. Maybe other people can contribute ideas as well.

Reply:
Maybe we can go with the name `CONFIG_INITIAL_PID_HASH_BITS` with a minimum value of `1` and init value `1u << CONFIG_INITIAL_PID_HASH_BITS`? So `2` hash bits will lead to value `4` that is capable of addressing pid values `0`, `1`, `2` and `3`.

Reply:
@pkarashchenko , I feel like shor and direct names as they are simpler for users. Actually the length doesn't have to be power-of-two, it is just a requirement from current hashing design. Direct names also last longer -- so even if later we drop the power-of-two requirements, names like `INITIAL_PID_LIMIT` still work.

But for `#warning` vs `static_assertion`. I need a clear direction from  you  and @xiaoxiang781216.

Reply:
Ok. If `INITIAL_PID_LIMIT` in general should not be a power of `2` and this is a limitation of the current hash algo then why do we need the compile time error? Why can't we simply round up `INITIAL_PID_LIMIT` to the nearest power of `2` and use it?

Reply:
I mean change `nxsched_npidhash() = 4;` to `nxsched_npidhash() = 1 << LOG2_CEIL(CONFIG_INITIAL_PID_LIMIT);`

Reply:
@pkarashchenko thanks again! this is really fruitful. I will send an update later.

Reply:
@pkarashchenko have you checked `LOG2_CEIL()` and `LOG2_FLOOR()`?
I got  5 for `LOG2_CEIL(16)` and 4 for `LOG2_FLOOR(16)`, so 16 becomes 32 using above formula!